src/main/java/org/commonjava/storage/pathmapped/core/FileBasedPhysicalStore.java

```diff
@@ -53,7 +53,13 @@ public OutputStream getOutputStream( FileInfo fileInfo ) throws IOException
     @Override
     public InputStream getInputStream( String storageFile ) throws IOException
     {
-        return new FileInputStream( new File( baseDir, storageFile ) );
+        File f = new File( baseDir, storageFile );
+        if ( !f.exists() )
```

Comment:
Should we also check if file is a regular file(not directory or symbol link)?

Reply:
Good idea. I will add !isDir check.

Reply:
Updated.